api/impl/daemon.go

```diff
@@ -91,57 +90,42 @@ func (nd *nodeDaemon) Init(ctx context.Context, opts ...api.DaemonInitOpt) error
 
 	initopts = append(initopts, node.AutoSealIntervalSecondsOpt(cfg.AutoSealIntervalSeconds))
 
-	if cfg.WithMiner != (address.Address{}) {
-		newConfig := rep.Config()
-		newConfig.Mining.MinerAddress = cfg.WithMiner
-		if err := rep.ReplaceConfig(newConfig); err != nil {
-			return err
-		}
+	if cfg.DevnetTest && cfg.DevnetNightly || cfg.DevnetTest && cfg.DevnetUser || cfg.DevnetNightly && cfg.DevnetUser {
```

Comment:
The associativity here isn't clear (I forget the rule of && and ||s). Please parenthesize.

Reply:
done and done